apps/condo/domains/resident/tasks/notifyResidentsOnPayday.js

```diff
@@ -0,0 +1,123 @@
+
+
+
+const logger = getLogger('notifyResidentsOnPayday')
+
+const DV_SENDER = { dv: 1, sender: { dv: 1, fingerprint: 'notifyResidentsOnPayday' } }
+const conflictingMessageTypes = [BILLING_RECEIPT_ADDED_TYPE, BILLING_RECEIPT_ADDED_WITH_DEBT_TYPE, BILLING_RECEIPT_ADDED_WITH_NO_DEBT_TYPE, BILLING_RECEIPT_AVAILABLE_NO_ACCOUNT_TYPE, BILLING_RECEIPT_AVAILABLE_TYPE, BILLING_RECEIPT_CATEGORY_AVAILABLE_TYPE, NEED_TO_PAY_RECEIPT_ON_PAYDAY_MESSAGE_TYPE]
+
+async function hasConflictingPushes (context, userId) {
+    const now = dayjs()
+    const userMessages = await Message.getAll(context, {
+        user: { id: userId },
+        createdAt_gte: now.set('hour', 0).set('minute', 0).set('second', 0).toISOString(),
+        createdAt_lte: now.toISOString(),
+    })
+
+    for (const message of userMessages) {
+        if (conflictingMessageTypes.find(type => type === message.type)) return true
+    }
+    return false
+}
+
+async function sendNotification (context, receipt) {
+    const serviceConsumers = await ServiceConsumer.getAll(context, {
+        accountNumber: get(receipt, 'account.number', null),
+        deletedAt: null,
+    })
+
+    for (const consumer of serviceConsumers) {
+        const resident = consumer.resident
+        const userId = resident.user.id
+        const lang = get(COUNTRIES, [get(receipt, 'organization.country', conf.DEFAULT_LOCALE), 'locale'], conf.DEFAULT_LOCALE)
+        const now = dayjs().subtract(1, 'month')
+
+        if (await hasConflictingPushes(context, userId)) {
+            logger.info({ msg: 'Today the User received push notifications with a conflicting type', userId })
+            continue
+        }
+        
+        const uniqKey = `${userId}_notification_on_payday`
+        await sendMessage(context, {
+            ...DV_SENDER,
+            to: { user: { id: userId } },
+            lang: conf.DEFAULT_LOCALE,
+            type: NEED_TO_PAY_RECEIPT_ON_PAYDAY_MESSAGE_TYPE,
+            meta: {
+                dv: 1,
+                data: {
+                    monthName: lang === RU_LOCALE ? now.locale('ru').format('MMMM') : now.format('MMMM'),
+                    billingReceiptId: { required: true },
+                    serviceConsumerId: consumer.id,
+                    residentId: resident.id,
+                    userId,
+                    toPayAmount: receipt.toPay,
+                },
+            },
+            uniqKey,
+        })
+    }
+
+}
+
+async function notifyResidentsOnPayday () {
+    const { keystone: context } = await getSchemaCtx('User')
+    const now = dayjs()
+    const currentPeriod = now.set('date', 1).format('YYYY-MM-DD')
+    const previousPeriod = now.set('date', 1).subtract(1, 'month').format('YYYY-MM-DD')
+    logger.info({ msg: 'Start proceeding', startAt: now.format() })
+
+    // get all BillingReceipts with positive toPay field
+    const allBillingReceipts = (await loadListByChunks({
```

Comment:
It looks so heavy! It's better to add this logic near the new receipt notification detection if it possible